### ISSUE TITLE
OpenAPI is always showing GET/POST/DELETE/PATCH on tables/views despite permissions

### ISSUE THREAD

**Issue by FrankvdAa** [PostgREST/postgrest#1870]
If a certain user isn't granted access to a certain table or view, it's not shown in the OpenAPI output, which seems correct.

But, views only allow for SELECT queries so why showing also POST/DELETE/PATCH options? Also, if I only allow certain access to a table, I would only expect the corresponding features in OpenAPI output.

Is there a reason that the OpenAPI is always showing GET/POST/DELETE/PATCH features for all tables and views?

**Comment by wolfgangwalther** [comment]
> But, views only allow for SELECT queries

This is not correct. See https://www.postgresql.org/docs/current/rules-views.html#RULES-VIEWS-UPDATE. Views can be made to support `INSERT`, `UPDATE` and `DELETE`.

> Is there a reason that the OpenAPI is always showing GET/POST/DELETE/PATCH features for all tables and views?

The only reason is that a more fine-grained output is not implemented, yet. We are unlikely to make major changes to the OpenAPI output right now, because we expect to move it out of core alltogether (see #1698). Once we have solved that, we're in a much better position to improve OpenAPI a lot.

**Comment by FrankvdAa** [comment]
> This is not correct. See https://www.postgresql.org/docs/current/rules-views.html#RULES-VIEWS-UPDATE. Views can be made to support `INSERT`, `UPDATE` and `DELETE`.

Good to know, wasn't aware of that. I'll then just have to wait for #1698 to be completed.

Thanks!

**Comment by laurenceisla** [comment]
Btw, here are the steps to reproduce this: https://github.com/PostgREST/postgrest/issues/2443#issue-1346175376

**Comment by vardhans07** [comment]
Hi! I'm new to open source and interested in contributing to PostgREST. This issue looks like a great starting point. I'd love to work on making the OpenAPI output reflect actual permissions. Could someone guide me on where to begin or which files to look at?

**Comment by laurenceisla** [comment]
Hi @vardhans07! Thanks for the initiative, but this issue is not so trivial to be supported in the main repo. We're moving away the OpenAPI implementation to [postgrest-openapi](https://github.com/PostgREST/postgrest-openapi), so our efforts will be in implementing the feature there. The permissions are not implemented at all in that repo, it would be quite a bit of work so wouldn't recommend to start there (you're free to check it out though). We have another [PostgREST issues](https://github.com/PostgREST/postgrest/issues) and you can check the ones with the `beginner` label which should be the easier ones to start with.
